### PR TITLE
Case Claim Title translation Python and Javascript implementation

### PR DESCRIPTION
**Pull request by dominicochoa-dimagi**
## Product Description

This is the python and javascript implementation of the case claim title translation. These changes include:
- updating suite_xml generation
- adding UI to 'Search and Claim Options' in case list config (see below)
Screen Shot 2022-05-01 at 4 36 11 PM
- adding title_label translation to app_manager bulk translations
- updating relevant tests

## Technical Summary

[Jira](https://dimagi-dev.atlassian.net/browse/USH-1824?atlOrigin=eyJpIjoiYmY3NGQ2OGE5NGIzNGVjNzhiZjUxMjE2MDdlZWExNWIiLCJwIjoiaiJ9)

## Feature Flag

CASE_CLAIM
### Safety story

These changes already have pretty good test coverage and have additional coverage from some changes I made.
### QA Plan

Will be QA'd after the Java implementation changes.

https://dimagi-dev.atlassian.net/browse/QA-4317

### Rollback instructions

- [x] This PR can be reverted after deploy with no further considerations

### Labels & Review
- [x] Risk label is set correctly
- [x] The set of people pinged as reviewers is appropriate for the level of risk of the change

### CODE REVIEW

File: corehq/apps/app_manager/tests/test_suite_registry_search.py [hunk 1]
```diff
@@ -46,7 +46,7 @@ class RemoteRequestSuiteTest(SimpleTestCase, TestXmlMixin, SuiteMixin):
     def setUp(self):
         self.app = Application.new_app(DOMAIN, "Untitled Application")
         self.app._id = '123'
-        self.app.build_spec = BuildSpec(version='2.53.0', build_number=1)
+        self.app.build_spec = BuildSpec(version='2.52.0', build_number=1)
```
Comment (AddisonDunn) [inline corehq/apps/app_manager/tests/test_suite_registry_search.py:49]: I thought registries/case search were 2.53-specific, but according to @orangejenny 's spec it seems like that's not the case?

File: corehq/apps/app_manager/id_strings.py [hunk 2]
```diff
@@ -86,8 +86,10 @@ def _referral_detail_title_locale():
 
 
 @pattern('m%d.%s.title')
-def detail_title_locale(detail_type):
+def detail_title_locale(app, module, detail_type):
```
[the remark this answers is not in the document]
Comment (orangejenny) [reply corehq/apps/app_manager/id_strings.py:89]: The thing that's a little off is that `module` implies `app`, so you wouldn't typically pass both of them, you'd just pass `module` and then use `module.get_app()` in the function to get access to the app.

File: corehq/apps/app_manager/templates/app_manager/partials/modules/case_search_properties.html [hunk 3]
```diff
@@ -156,6 +156,16 @@ <h4 class="panel-title panel-title-nolink">{% trans "Search and Claim Options" %
               {% include "app_manager/partials/nav_menu_media.html" with qualifier='case_search-search_again_label_media_' ICON_LABEL="Search Again Icon" AUDIO_LABEL="Search Again Audio" %}
             <!-- /ko -->
           </div>
+          <div class="form-group">
+            <label for="case_claim_title" class="{% css_label_class %} control-label">
+              {% trans "Case Claim Title" %}
```
Comment (orangejenny) [inline corehq/apps/app_manager/templates/app_manager/partials/modules/case_search_properties.html:161]: How about "Search Screen Title"?

File: corehq/apps/app_manager/tests/test_suite_registry_search.py (hunk 1 above)
Comment (orangejenny) [reply corehq/apps/app_manager/tests/test_suite_registry_search.py:49]: Case search isn't, but registries are, yes, so this should indeed be 2.53. I'm surprised these tests are passing.

File: corehq/apps/app_manager/tests/test_suite_registry_search.py [hunk 4]
```diff
@@ -295,7 +296,7 @@ def setUp(self):
         super().setUp()
         self.app = Application.new_app(DOMAIN, "Application with Shadow")
         self.app._id = '456'
-        self.app.build_spec = BuildSpec(version='2.53.0', build_number=1)
+        self.app.build_spec = BuildSpec(version='2.52.0', build_number=1)
```
Comment (orangejenny) [inline corehq/apps/app_manager/tests/test_suite_registry_search.py:299]: Same as above, this should be 2.53.

File: corehq/apps/app_manager/tests/test_suite_remote_request.py [hunk 5]
```diff
@@ -788,3 +788,54 @@ def test_exclude_from_search(self, *args):
         </partial>
         """
         self.assertXmlPartialEqual(expected, suite, "./remote-request[1]/session/query/prompt[@key='name']")
+
+    def test_case_search_title_translation(self, *args):
```
Comment (orangejenny) [inline corehq/apps/app_manager/tests/test_suite_remote_request.py:792]: Nit: I'd split this into two tests and name them so it's clear that one is testing legacy behavior and the other is testing the new behavior.

File: corehq/apps/app_manager/tests/test_suite_remote_request.py [hunk 6]
```diff
@@ -788,3 +788,54 @@ def test_exclude_from_search(self, *args):
...
+        <partial>
+            <title>
+              <text>
+                <locale id="cchq.case"/>
+              </text>
+            </title>
+        </partial>
+        """
+        self.assertXmlPartialEqual(expected_remote_request, suite, "./remote-request[1]/command")
+        self.assertXmlPartialEqual(expected_search_detail, suite, "./detail[@id='m0_search_short']/title")
```
Comment (orangejenny) [inline corehq/apps/app_manager/tests/test_suite_remote_request.py:816]: You could consider asserting more broadly than this for the sake of picking up undesirable side effects - the entire detail, or even both the the m0_search_short and m0_case_short details.

File: corehq/apps/app_manager/views/modules.py [hunk 7]
```diff
@@ -240,6 +240,8 @@ def _get_shared_module_view_context(request, app, module, case_property_builder,
                 module.search_config.search_label.label if hasattr(module, 'search_config') else "",
             'search_again_label':
                 module.search_config.search_again_label.label if hasattr(module, 'search_config') else "",
+            'title_label':
+                module.search_config.title_label if hasattr(module, 'search_config') else "",
```
[the remark this answers is not in the document]
Comment (orangejenny) [reply corehq/apps/app_manager/views/modules.py:244]: This check has to do with module types. This function gets called for all modules, but a ReportModule wouldn't have a search_config attribute.

**Review by orangejenny — COMMENTED** [review]
looks good, just minor feedback

File: corehq/apps/app_manager/templates/app_manager/partials/modules/case_search_properties.html [hunk 8]
```diff
@@ -156,6 +156,16 @@ <h4 class="panel-title panel-title-nolink">{% trans "Search and Claim Options" %
...
+              <span class="hq-help-template" data-title="{% trans "Case Claim Title Translation" %}"
+                    data-content="{% trans_html_attr "Lorem Ipsum" %}">
+            </label>
+            <div class="{% css_field_class %}">
+              {% input_trans module.search_config.title_label langs input_name='title_label' input_id='title_label' data_bind="value: titleLabel" %}
+            </div>
+          </div>
```
[the remark this answers is not in the document]
Comment (AddisonDunn) [reply corehq/apps/app_manager/templates/app_manager/partials/modules/case_search_properties.html:168]: bump @dominicochoa-dimagi

[the remark this answers is not in the document]
Comment (AddisonDunn) [reply corehq/apps/app_manager/templates/app_manager/partials/modules/case_search_properties.html:168]: Oh I see you added it to the PR description. Thanks!

File: corehq/apps/app_manager/id_strings.py [hunk 9]
```diff
@@ -86,8 +86,10 @@ def _referral_detail_title_locale():
 
 
 @pattern('m%d.%s.title')
-def detail_title_locale(detail_type):
-    if detail_type.startswith('case') or detail_type.startswith('search'):
+def detail_title_locale(module, detail_type):
```
Comment (orangejenny) [inline corehq/apps/app_manager/id_strings.py:89]: Can't you remove these changes now that the translation is always available in the new display block?

File: corehq/apps/app_manager/suite_xml/post_process/remote_requests.py [hunk 10]
```diff
@@ -137,12 +137,20 @@ def get_post_relevant(self):
...
+                    text=Text(locale_id=id_strings.case_search_title_tranlsation(self.module)),
+                ),
+            )
+        else:
+            return Command(
+                id=id_strings.search_command(self.module),
+                display=Display(
+                    text=Text(locale_id=id_strings.case_search_locale(self.module)),
+                ),
+            )
```
[the remark this answers is not in the document]
Comment (orangejenny) [reply corehq/apps/app_manager/suite_xml/post_process/remote_requests.py:146]: this can also be removed now that you have `build_title`, right?

**Comment by orangejenny** [comment]
The new suite change (fb9be938e8e40ae77711ccbbb0a61d9b3f00363e) looks good. I think you can simplify/remove some of your previous suite code.

I'd expect that you'd also need to update tests (just about any suite change should have test coverage), but it looks like tests aren't running on this PR anymore. Merging master, just for the sake of adding a commit, might get them going again.

File: corehq/apps/app_manager/suite_xml/post_process/remote_requests.py [hunk 11]
```diff
@@ -140,6 +140,11 @@ def build_command(self):
             ),
         )
 
+    def build_title(self):
+        return Display(
```
Comment (orangejenny) [inline corehq/apps/app_manager/suite_xml/post_process/remote_requests.py:144]: This uses the `Display` class, but the XML in tests doesn't have a `<display>` block, do you know why that is?

**Review by orangejenny — COMMENTED** [review]
Re-reviewed from the top. Looks good, left a couple of questions.

**Comment by orangejenny** [comment]
(haven't forgotten about final review for this but might not get to it until Monday)

File: corehq/apps/app_manager/app_strings.py [hunk 12]
```diff
@@ -184,11 +184,15 @@ def maybe_add_index(text):
                     lang, for_default=for_default, build_profile_id=build_profile_id)
                 search_again_label_audio = module.search_config.search_again_label.audio_app_string(
                     lang, for_default=for_default, build_profile_id=build_profile_id)
+                title_label = module.search_config.get_search_title_label(app, lang, for_default=for_default)
```
Comment (orangejenny) [inline corehq/apps/app_manager/app_strings.py:187]: Nitpick: move this into the `if` below. It's good to reduce variable [live time](http://www.rosscode.com/blog/index.php?title=variable_usage_span_and_live_time&more=1&c=1&tb=1&pb=1).

**Review by orangejenny — APPROVED** [review]

**Pull request merged**